scipy/stats/_continuous_distns.py

```diff
@@ -4189,6 +4189,50 @@ def _stats(self):
     def _entropy(self):
         return np.log(2*np.pi)
 
+    @_call_super_mom
+    @inherit_docstring_from(rv_continuous)
+    def fit(self, data, *args, **kwds):
+        if kwds.pop('superfit', False):
+            return super().fit(data, *args, **kwds)
+
+        data, floc, fscale = _check_fit_input_parameters(self, data,
+                                                         args, kwds)
+
+        # location is independent from the scale
+        data_min = np.min(data)
+        if floc is not None:
+            if data_min < floc:
+                # There are values that are less than the specified loc.
+                raise FitDataError("halfcauchy", lower=floc, upper=np.inf)
+            loc = floc
+        else:
+            # if not provided, location MLE is the minimal data point
+            loc = data_min
+
+        # find scale
+        def find_scale(loc, data):
+            shifted_data = data - loc
+            n = data.size
+            shifted_data_squared = np.square(shifted_data)
+            x0 = np.median(shifted_data)
+
+            def fun_to_solve(scale):
+                denominator = scale**2 + shifted_data_squared
+                f = 2 * np.sum(shifted_data_squared/denominator) - n
+                grad = - 4 * np.sum(shifted_data_squared * scale
+                                    /denominator**2)
+                return f, grad
+
+            res = root_scalar(fun_to_solve, fprime=True, x0=x0)
```

Comment:
```suggestion
                return 2 * np.sum(shifted_data_squared/denominator) - n

            small = np.finfo(1.0).tiny**0.5
            res = root_scalar(fun_to_solve, bracket=(small, np.max(shifted_data)))
```

Reply:
Ah, and after removing the `median` calculation, it's faster still:

<img width="852" alt="image" src="https://github.com/scipy/scipy/assets/6570539/70b3791b-5fa7-4238-b343-f3a1217c76e8">